addons/mail/models/mail_thread.py

```diff
@@ -1181,7 +1183,10 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
 
                 mail_message = self.env['mail.message']
                 if email_part:
-                    email = email_part.get_payload()[0]
+                    if email_part.is_multipart():
```

Comment:
In this particular case explicitly testing for content-type would be more readable and future-proof wrt the intent:
```suggestion
                    if email_part.get_content_type() == 'message/rfc822':
                        # message/rfc822 is multipart with sub-messages
```